### ISSUE TITLE
Scripts compiles in every execution even if unchanged

### ISSUE THREAD

**Issue by DavidPerezIngeniero** [lihaoyi/Ammonite#568]
Install Ammonite in a new computer, but don't execute.

Create any Ammonite script with this header:
```
#!/usr/bin/env amm
```

The script compiles in *every* execution, until you don't launch the `amm` binary and compiles the needed `*.sc`:
```
$ amm
Loading...
Compiling replBridge.sc
Compiling interpBridge.sc
Compiling HardcodedPredef.sc
Compiling ArgsPredef.sc
Compiling predef.sc
Compiling SharedPredef.sc
Compiling LoadedPredef.sc
Welcome to the Ammonite Repl 0.8.1
(Scala 2.12.1 Java 1.8.0_111)
```

**Comment by DavidPerezIngeniero** [comment]
Apart from that, I've observed that many scripts with $ivy imports, compile in every execution, even if unchanged.  Using latest Ammonite.

**Comment by lihaoyi** [comment]
I can't reproduce this behavior. In fact, trying to run that script file using the latest version of Ammonite was crashing with a stack trace! I fixed the stack trace with 642af7c25433bf9e97968656a0fb724c8dcca808, and 5ad6ad193fc0d67ecdb2f544e417afaa3bdc9bd8 fixes some problems around unnecessarily/incorrectly recompiling files, so hopefully this problem should go away...

Compiling info messages going to stdout and screwing things up can be avoided with `--silent`/`-s`, and improving that is tracked in https://github.com/lihaoyi/Ammonite/issues/574

**Comment by DavidPerezIngeniero** [comment]
I've tried Ammonite 0.8.5 and it continues to recompile after every execution.
I'll create a new issue
